packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -1437,6 +1437,25 @@ function getDependency(node) {
     )
   ) {
     return getDependency(node.parent);
+  } else {
+    return stripTailingPropInAssignment(node);
```

Comment:
What makes you think it's an assignment in this branch? Can we make it more specific and throw on something unexpected?

Reply:
😅  `tryStrippingTrailingPropsInAssigment` could have been a better choice. I'm trying to enumerate all cases and confused by a test:

```js
function MyComponent(props) {
  useEffect(() => {
    if (props.foo.onChange) {
      props.foo.onChange();
    }
  }, []);
}
```

Why is `props.foo` a proper dependency? Why not `props.foo.onChange`?